src/app/file-manager/file-explorer/file-explorer.component.ts

```diff
@@ -1,37 +1,48 @@
-import { Component, OnInit, NgZone, ViewChild } from '@angular/core';
+import { Component, OnInit, NgZone, ViewChild, AfterViewInit } from '@angular/core';
 import { Item } from '../models/item';
-import { BucketObjectsDB } from '../models/bucket-object-db';
+import { SelectionStatus } from '../models/selection-status';
+
 import { FilterSizePipe } from '../filters/filesize-filter';
 import { FileDownloadModalComponent } from '../file-download-modal/file-download-modal.component';
-import { MatDialog, MatPaginator } from '@angular/material';
+import { MatDialog, MatPaginator, MAT_CHECKBOX_CLICK_ACTION } from '@angular/material';
+import { MatTableDataSource, MatSort } from '@angular/material';
 import { StatusService } from '../services/status.service';
 import { BucketService } from '../services/bucket.service';
-import { SelectionModel } from '@angular/cdk/collections';
 import { FirecloudApiService } from '@app/file-manager/services/firecloud-api.service';
 import { FilesDataSource } from '@app/file-manager/dbstate/files-datasource';
 import { FilesService } from '@app/file-manager/services/files.service';
 import { WarningModalComponent } from '@app/file-manager/warning-modal/warning-modal.component';
+import { ItemStatus } from '@app/file-manager/models/item-status';
+import { List } from 'lodash';
+
+import { NgxSpinnerComponent } from 'ngx-spinner';
+import { NgxSpinnerService } from 'ngx-spinner';
+import { applySourceSpanToStatementIfNeeded } from '@angular/compiler/src/output/output_ast';
+import { SelectionService } from '@app/file-manager/services/selection.service';
 
 @Component({
   selector: 'app-file-explorer',
   templateUrl: './file-explorer.component.html',
-  styleUrls: ['./file-explorer.component.scss']
+  styleUrls: ['./file-explorer.component.scss'],
+  providers: [
+    { provide: MAT_CHECKBOX_CLICK_ACTION, useValue: 'check-indeterminate' }
+  ]
 })
-export class FileExplorerComponent implements OnInit {
+export class FileExplorerComponent implements OnInit, AfterViewInit {
+
   isHome: Boolean = false;
-  bucketObjects = new BucketObjectsDB();
-  breadcrumbPath = '';
-  displayedColumns = ['select', 'name', 'size', 'lastdate'];
-  dataSource: FilesDataSource | null;
-  selection = new SelectionModel<Item>(true, []);
+  displayedColumns = ['select', 'displayName', 'size', 'updated'];
+
+  dataSource = new MatTableDataSource([]);
+  @ViewChild(MatSort) sort: MatSort;
   @ViewChild(MatPaginator) paginator: MatPaginator;
+
+  breadcrumbItems: Item[];
```

Comment:
I'm confused why we have `breadcrumbItems` here as well as in the breadcrumb component above.

Reply:
breadcrumbItems array here holds the items:Item[] to be passed to bredcrumb components.

Reply:
OK, I see now. Comments would be helpful.